src/Http/Http.Results/src/Results.cs

```diff
@@ -504,24 +527,47 @@ public static IResult UnprocessableEntity(object? error = null)
         /// <param name="statusCode">The status code.</param>
         /// <param name="title">The value for <see cref="ProblemDetails.Title" />.</param>
         /// <param name="type">The value for <see cref="ProblemDetails.Type" />.</param>
+        /// <param name="extensions">The value for <see cref="ProblemDetails.Extensions" />.</param>
         /// <returns>The created <see cref="IResult"/> for the response.</returns>
         public static IResult ValidationProblem(
             IDictionary<string, string[]> errors,
             string? detail = null,
             string? instance = null,
             int? statusCode = null,
             string? title = null,
-            string? type = null)
+            string? type = null,
+            IDictionary<string, object?>? extensions = null)
         {
             var problemDetails = new HttpValidationProblemDetails(errors)
             {
                 Detail = detail,
                 Instance = instance,
                 Title = title,
                 Type = type,
-                Status = statusCode,
+                Status = statusCode
             };
 
+            if (extensions is not null)
+            {
+                foreach (var extension in extensions)
+                {
+                    problemDetails.Extensions.Add(extension);
+                }
+            }
+
+            return new ObjectResult(problemDetails)
+            {
+                ContentType = "application/problem+json",
+            };
+        }
+
+        /// <summary>
+        /// Produces a <see cref="HttpValidationProblemDetails"/> response.
+        /// </summary>
+        /// <param name="problemDetails">The <see cref="HttpValidationProblemDetails"/>  object to produce a response from.</param>
+        /// <returns>The created <see cref="IResult"/> for the response.</returns>
+        public static IResult ValidationProblem(HttpValidationProblemDetails problemDetails)
```

Comment:
Do we need this overload? `Results.Problem(..)` works for all classes of problem details, so this feels gratuitous.

Reply:
I can't think of a need to have a specific overload for that flavor of problem details off the top of my head. What would it do differently? Ultimately doesn't it pass it to the code that specializes anyway?

Reply:
@halter73 pointed out it's mostly about discoverability i.e. you have a validation problem so that's what you search for. But it's very unusual to add an APIs primarily as a convenience / alias.